src/bases/createEvented.ts

```diff
@@ -61,6 +61,26 @@ function handlesArraytoHandle(handles: Handle[]): Handle {
 		}
 	};
 }
+/**
+ * Internal function to check if a target string matches a glob string that contains wildcards.
+ * Note: Due to limited use cases in event type name, currently only `*` that matches 0 or more characters is supported.
+ *
+ * @param globString The glob string that contains wildcards pattern
+ * @param targetString The string under test
+ * @return boolean match result
+ */
+
+function isGlobMatch(globString: string, targetString: string): boolean {
+	if (!globString || !targetString) {
+		return false;
+	}
+	const regex = new RegExp(`^${ globString.replace(/\*/g, '.*') }$`);
+
+	if (regex.test(targetString)) {
```

Comment:
Thanks @agubler for the review. I agree that creating `RegExp` everytime is expensive and I agree pre-compiling it inside `on` would be a better solution. However, currently `dojo/core` event system seems to only work with `string` as the `type/methodName`, and this can't be easily changed. For example, if I change the [`methodName` in `getDispatcher()`](https://github.com/dojo/core/blob/40326b799b462da9d5716507398f0a2632c15fda/src/aspect.ts#L138) to be a `string | Regex`, it won't work with [`Indexable` whose keys need to be a `string` only](https://github.com/dojo/core/blob/40326b799b462da9d5716507398f0a2632c15fda/src/aspect.ts#L22). Getting rid of `Indexable` seems not an option even though it will make this work.

As another solution, I added a internal map to cache created `RegExp` objects.

Please let me know if you have any questions or other concerns. Thanks!